src/msg/simple/Pipe.cc

```diff
@@ -2565,18 +2594,20 @@ ssize_t Pipe::buffered_recv(char *buf, size_t len, int flags)
 
 ssize_t Pipe::tcp_read_nonblocking(char *buf, unsigned len)
 {
+  int rc = -1;
   ssize_t got = buffered_recv(buf, len, MSG_DONTWAIT );
```

Comment:
same thing here, the errno value is only useful all the way in do_recv here, buffered_recv and tcp_read_nonblocking should just pass the return code up, and not read errno themselves, since they are not calling any syscalls that set it